pipes.go

```diff
@@ -0,0 +1,38 @@
+package command
+
+import (
+	"bufio"
+	"io"
+)
+
+// Pipes represents the trinity of standard io pipes
+// under POSIX.
+type Pipes struct {
+	Stdin  io.WriteCloser
+	Stdout io.Reader
+	Stderr io.Reader
+}
+
+// Pipes Returns the raw types object, because
+// this may be something we want to make by scratch
+// in a future type, such as in situations where you
+// filter/wrap output.
+func (p *Pipes) Pipes() *Pipes {
+	return p
+}
+
+// StdoutScanner returns a bufio.Scanner over stdio.
+func (p *Pipes) StdoutScanner() *bufio.Scanner {
```

Comment:
would it be more appropriate to call this `NewStdoutScanner()`?

My first experiment with the interface without reading the source code carefully I thought this was just returning an already initialized Scanner so i did...

```
for p.StdoutScanner().Scan() {
			text := p.StdoutScanner().Text()
   ...
```

Reply:
`*(exec.Cmd).StdoutPipe()` is the pattern I'm following, which *also* creates a new thing.

In both cases, having two copies would make no sense. So my fix would probably be to add two private sync.RunOnce types, and returning error for the second request.

Reply:
gotcha, would the interface then change to *bufioScanner, error I guess?

yeah i can't decide if that would be better, or just adding a little more in the docstring. like

the most verbose I could think of would be something like

"StdoutScanner returns a newly initialized bufio.Scanner over stdio, should only be called once"